framework/wazuh/core/cluster/cluster.py

```diff
@@ -316,42 +362,71 @@ async def async_decompress_files(zip_path, ko_files_name="files_metadata.json"):
     return decompress_files(zip_path, ko_files_name)
 
 
-def decompress_files(zip_path, ko_files_name="files_metadata.json"):
-    """Unzip files in a directory and load the files_metadata.json as a dict.
+def decompress_files(compress_path, ko_files_name="files_metadata.json"):
+    """Decompress files in a directory and load the files_metadata.json as a dict.
+
+    To avoid consuming too many memory resources, the compressed file is read in chunks
+    of 'windows_size' and split based on a file separator.
 
     Parameters
     ----------
-    zip_path : str
-        Full path to the zip file.
+    compress_path : str
+        Full path to the compress file.
     ko_files_name : str
-        Name of the metadata json inside zip file.
+        Name of the metadata json inside the compress file.
 
     Returns
     -------
     ko_files : dict
         Paths (keys) and metadata (values) of the files listed in cluster.json.
     zip_dir : str
-        Full path to unzipped directory.
+        Full path to decompressed directory.
     """
+    ko_files = ''
+    compressed_data = b''
+    window_size = 1024*1024*10   # 10 MiB
+    decompress_dir = compress_path + 'dir'
```

Comment:
```suggestion
    decompress_dir = path.join(compress_path, 'dir')
```

Reply:
`dir` is added to create a directory with the same name as the compress. The below can be found while decompressing, there are a compressed file (first one) and a compressdir:
```
root@wazuh-worker1:/var/ossec/queue/cluster/worker1# ls
worker1-1644485176.480029-9301578326898758.compressed
worker1-1644485176.480029-9301578326898758.compresseddir
```

Since a file with that name already exists, a directory with the same name cannot be created:
```
worker1-1644485176.480029-9301578326898758.compressed (file)
worker1-1644485176.480029-9301578326898758.compressed/dir (directory)
```